File: 2019/4/pt2.rb
s,f = ARGF.read().strip.split('-') #Args
p (s...f).each.
    select{|s|s==s.chars.sort.join}. #Sorted integers?
    select{|s|
        s.chars.uniq
            map{|c|s.count(c)}
            any?{|c|c==2}  #Doubles?
    }.length

File: 2019/4/pt2.py
start, end = map(int, open("input").read().split("-"))
ordered = lambda s: list(s) == sorted(s)
paired = lambda s: any(s.count(c) == 2 for c in set(s))
valid = lambda s: ordered(s) and paired(s)
print(sum(1 for _ in filter(valid, map(str, range(start, end)))))
